### PR TITLE
feat(CryptoCurrencies): add crypto currencies and wallets

### PR DESCRIPTION
**Pull request by gabriele-v**

### CODE REVIEW

**Comment by slodki** [comment]
Have we any procedure to bump database version correctly with both desktop and android app?

**Comment by gabriele-v** [comment]
I don't know exactly how it works in Android version but yes, we have migrated to new sql files upgrade scheme exactly to keep in sync Desktop and Android.

But in this Issue there is a table that doesn't exist in desktop version:
https://github.com/moneymanagerex/android-money-manager-ex/issues/1139

@MisterY how could this be possible?
Android DB should be at version 7 like desktop, upgrade of this PR should be version 8, could you confirm?

**Comment by MisterY** [comment]
@gabriele-v This may have to do with the code propagation. I take the db code from the db repository, meaning I'm not sure in which desktop release it will appear. I was expecting the master to be the stable branch and the code gets committed when it is ready for publishing.
I've also seen some strange cases, but not enough to figure out a pattern and see whether the upgrade works in all cases.

File: database_version_8.sql [hunk 1]
```diff
@@ -0,0 +1,3 @@
+alter table CURRENCYFORMATS_V1 add column CURRENCY_TYPE TEXT;
+update CURRENCYFORMATS_V1 set CURRENCY_TYPE = 'Traditional';
```
Comment (vomikan) [inline database_version_8.sql]: https://en.m.wikipedia.org/wiki/Fiat_money
I suggest to use more simple string for non crypto currencied : 'fiat'

**Pull request merged**